src/Stratis.Bitcoin.Features.Miner.Tests/Controllers/MiningControllerTest.cs

```diff
@@ -180,5 +181,54 @@ public void StopMining_ReturnsSuccess()
             var okResult = Assert.IsType<OkResult>(result);
             Assert.NotNull(okResult);
         }
+
+        [Fact]
+        public void MiningController_Calls_GetAccount_Returns_Reference_To_Wallet_Account()
+        {
+            Mock<IConsensusManager> consensusManager = new Mock<IConsensusManager>();
+            Mock<IPowMining> powMining = new Mock<IPowMining>();
+
+            var walletManager = new Mock<IWalletManager>();
+            walletManager.Setup(f => f.GetWalletsNames()).Returns(new List<string> { wallet });
+            walletManager.Setup(f => f.GetAccounts(wallet)).Returns(new List<HdAccount> { new HdAccount { Name = account } });
+
+            this.fullNode.Setup(f => f.NodeFeature<MiningFeature>(false)).Returns(new MiningFeature(this.network, new MinerSettings(Configuration.NodeSettings.Default(this.network)), Configuration.NodeSettings.Default(this.network), this.loggerFactory, new Mock<ITimeSyncBehaviorState>().Object, powMining.Object, null));
+
+            MiningController controller = new MiningController(consensusManager.Object, this.fullNode.Object, this.loggerFactory, this.network, new Mock<IPowMining>().Object, walletManager.Object);
+            WalletAccountReference walletAccountReference = controller.GetAccount();
+
+            Assert.False(string.IsNullOrEmpty(walletAccountReference.AccountName));
```

Comment:
Might as well check that it's the right account that is returned.

```suggestion
            Assert.Equal(account, walletAccountReference.AccountName));
```